getting-started/README.md

```diff
@@ -0,0 +1,264 @@
+
+#### Truffle
+
+In [Truffle](https://trufflesuite.com/):
+
+1. Add the `@truffle/hdwallet-provider` package:
+
+   ```sh
+   yarn add @truffle/hdwallet-provider
+   ```
+
+1. Edit the `truffle-config.js` file
+
+   1. Uncomment this line:
+
+      ```js
+      const HDWalletProvider = require('@truffle/hdwallet-provider')
+      ```
+
+   1. Edit `modules.export.networks` to add a definition similar to this one:
+
+      ```js 
+      "optimistic-kovan": {
+         provider: () => new HDWalletProvider(<your mnemonic>, <Optimism URL>)
+      }
+      ```
+
+## Interacting with Optimism contracts
+
+We have [Hardhat's Greeter contract](https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-core/sample-projects/basic/contracts/Greeter.sol) on Optimistic Kovan, at address [0xE0A5fe4Fd70B6ea4217122e85d213D70766d6c2c](https://kovan-optimistic.etherscan.io/address/0xe0a5fe4fd70b6ea4217122e85d213d70766d6c2c). 
+You can verify your development stack configuration by interacting with it.
+To see it in action, follow the instructions below for your development stack.
+
+
+### Hardhat
+
+1. Run the console:
+   ```sh
+   cd hardhat
+   yarn
+   yarn hardhat console --network optimistic-kovan
+   ```
+
+1. Connect to the Greeter contract:   
+
+   ```js
+   Greeter = await ethers.getContractFactory("Greeter")
+   greeter = await Greeter.attach("0xE0A5fe4Fd70B6ea4217122e85d213D70766d6c2c")
+   ```   
+
+1. Read information from the contact:
+
+   ```js
+   await greeter.greet()
+   ```
+
+1. Submit a transaction, wait for it to be processed, and see that it affected the state.
+   Note that the account used by default, [0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266](https://kovan-optimistic.etherscan.io/address/0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266), may not have enough ETH. 
+   In that case, edit the configuration file to use your own mnemonic.
+
+   ```js
+   tx = await greeter.setGreeting(`Hello ${new Date()}`)
+   receipt = await tx.wait()   // Doesn't work in Truffle
+   await greeter.greet()
+   ```
+
+### Truffle
+
+1. Install the software, compile the contract, and run the console:
+
+   ```sh
+   cd truffle
+   yarn
+   yarn truffle compile
+   yarn truffle console --network optimistic-kovan
+   ```
+
+1. Connect to the Greeter contact:
+
+   ```js
+   greeter = await Greeter.at("0xE0A5fe4Fd70B6ea4217122e85d213D70766d6c2c")
+   ```
+
+1. Read information from the contact:
+
+   ```js
+   await greeter.greet()
+   ```
+
+1. Submit a transaction, wait for it to be processed, and see that it affected the state.
+   Note that the account used by default, [0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266](https://kovan-optimistic.etherscan.io/address/0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266), may not have enough ETH. 
+   In that case, edit the configuration file to use your own mnemonic.
+
+   ```js
+   tx = await greeter.setGreeting(`Hello ${new Date()}`)
+   await greeter.greet()
+   ```
+
+
+### Remix
```

Comment:
Can we please combine these? feels weird to mention a framework (i.e hardhat) and then later in a new section explain how to use it.